test/preview/components/samplers/test_top_p.py

```diff
@@ -0,0 +1,78 @@
+import pytest
+
+from haystack.preview import Document, ComponentError
+from haystack.preview.components.samplers.top_p import TopPSampler
+
+
+class TestTopPSampler:
+    @pytest.mark.unit
+    def test_to_dict(self):
+        component = TopPSampler()
+        data = component.to_dict()
+        assert data == {"type": "TopPSampler", "init_parameters": {"top_p": 1.0, "score_field": None}}
+
+    @pytest.mark.unit
+    def test_to_dict_with_custom_init_parameters(self):
+        component = TopPSampler(top_p=0.92)
+        data = component.to_dict()
+        assert data == {"type": "TopPSampler", "init_parameters": {"top_p": 0.92, "score_field": None}}
+
+    @pytest.mark.unit
+    def test_from_dict(self):
+        data = {"type": "TopPSampler", "init_parameters": {"top_p": 0.9, "score_field": None}}
+        component = TopPSampler.from_dict(data)
+        assert component.top_p == 0.9
+
+    @pytest.mark.unit
+    def test_run_scores_from_metadata(self):
+        """
+        Test if the component runs correctly with scores already in the metadata.
+        """
+        sampler = TopPSampler(top_p=0.95, score_field="similarity_score")
+        docs = [
+            Document(text="Berlin", metadata={"similarity_score": -10.6}),
+            Document(text="Belgrade", metadata={"similarity_score": -8.9}),
+            Document(text="Sarajevo", metadata={"similarity_score": -4.6}),
+        ]
+        output = sampler.run(documents=docs)
+        docs = output["documents"]
+        assert len(docs) == 1
+        assert docs[0].text == "Sarajevo"
```

Comment:
Similar as to the SimilarityRanker I would add an assert checking that the expected order is returned. I would even mess up the order of the input docs to see if the sorting within this component is working as expected. What do you think?

Reply:
@vblagoje just this last comment about tests, especially since this caught a bug in the last PR 😅